network-api/networkapi/wagtailpages/pagemodels/customblocks/current_events_slider_list_block.py

```diff
@@ -0,0 +1,73 @@
+from wagtail.core import blocks
+from wagtail.images.blocks import ImageChooserBlock
+
+
+class CurrentEventBaseButtonBlock(blocks.StructBlock):
+    label = blocks.CharBlock(help_text='Button label')
+
+
+class CurrentEventButtonInternalPage(CurrentEventBaseButtonBlock):
+    link = blocks.PageChooserBlock(help_text='Page that this button should link out to.')
+
+
+class CurrentEventButtonExternalPage(CurrentEventBaseButtonBlock):
+    link = blocks.URLBlock(help_text='URL that this button should link out to.')
```

Comment:
These are such generic things they can probably be shared blocks in `utils/blocks` or something like that. I find it highly unlikely these will change and if we make these for every card block we create in MozFest, one day we're just going to have:

- `CurrentEventBaseButtonBlock`, `CurrentEventButtonInternalPage`, `CurrentEventButtonExternalPage`
- `PastEventBaseButtonBlock`, `PastEventButtonInternalPage`, `PastEventButtonExternalPage`
- `FutureEventBaseButtonBlock`, `FutureEventButtonInternalPage`, `FutureEventButtonExternalPage`
- `RelatedEventBaseButtonBlock`, `RelatedEventButtonInternalPage`, `RelatedEventButtonExternalPage`
- `BaseButtonBlock`, `NewsButtonInternalPage`, `NewsButtonExternalPage`
- `RelatedBaseButtonBlock`, `RelatedNewsButtonInternalPage`, `RelatedNewsButtonExternalPage`

which seems a bit excessive and repetitive.

I think we can also add the `link_url` property to `CurrentEventBaseButtonBlock` instead of having a completely separate `CurrentEventBlockStructValue` with `buttons`? Feel free to correct me if this one's not viable in some way.